platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -284,15 +286,39 @@ private void captureTimestamp(final ConsensusEvent event) {
      * @param transaction the transaction to apply
      */
     private void handleTransaction(final ConsensusTransaction transaction) {
-        if (transaction.isSystem()) {
-            return;
+        int delta;
+
+        if (!transaction.getTransaction().equals(EventTransaction.DEFAULT)) {
+            if (transaction.isSystem()) {
+                return;
+            }
+
+            delta = ByteUtils.byteArrayToInt(
+                    transaction.getApplicationTransaction().toByteArray(), 0);
+
+        } else {
+            if (isSystemTransaction(transaction.getTransactionsBytes())) {
+                return;
+            }
+
+            delta = ByteUtils.byteArrayToInt(transaction.getTransactionsBytes().toByteArray(), 0);
         }
-        final int delta =
-                ByteUtils.byteArrayToInt(transaction.getApplicationTransaction().toByteArray(), 0);
+
         runningSum += delta;
         setChild(RUNNING_SUM_INDEX, new StringLeaf(Long.toString(runningSum)));
     }
 
+    // TODO Temporary method to filter out system transactions, so that platform is tested with application and system
+    // transactions. Once callbacks are introduced this method won't be needed.
```

Comment:
I agree. My main point here is that the logic for determining which transactions are system and which are not will be changed once we have `hedera.app` deserializing and filtering the system transactions in the callback.

I will remove the comment so that it's not confusing readers of the code.